public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -54,9 +55,9 @@ export const GeneratePopoverBody: React.FC<{
 
   useEffect(() => {
     const getMonitorType = async () => {
-      const context = await incontextInsight.contextProvider?.();
-      const monitorType = context?.additionalInfo?.monitorType;
-      const dsl = context?.additionalInfo?.dsl;
+      if (!contextObject) return;
```

Comment:
Nit: The `contextObject` was initialized in `onGenerateSummary` below and was only executed once. The `displayDiscoverButton` won't be true if `contextObject` is undefined. I'm not sure if it matches the business logic.